### PR TITLE
Add AOT flag to worker template

### LINKED ISSUE #46541: Add AOT flag to worker template
The worker template is compatible with native AOT. Should there be a flag in the template to enable publishing as AOT?

### PR DESCRIPTION
**Pull request by eerhardt**
The `-aot` flag is available in the `api`, `grpc`, and soon to be in the `console` templates. The `worker` template is compatible with NativeAOT, so it should have the option as well.

Fix #46541

### CODE REVIEW

**Comment by JamesNK** [comment]
Are there template tests for worker? Should add aot publish

**Comment by eerhardt** [comment]
> Are there template tests for worker? Should add aot publish

Done. I left them skipped for now until https://github.com/dotnet/aspnetcore/pull/47247/ is fixed.

File: src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public async Task ApiTemplateCSharp()
         await ApiTemplateCore(languageOverride: null);
     }
 
-    [ConditionalFact(Skip = "Unskip when there are no more build or publish warnings for native AOT.")]
+    [ConditionalFact(Skip = "Unskip when Helix supports native AOT. https://github.com/dotnet/aspnetcore/pull/47247/")]
```
Comment (captainsafia) [inline src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs:38]: Nit: Should we update the tests in the gRPC templates with an issue reference as well?

https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/test/Templates.Tests/GrpcTemplateTest.cs

**Review by captainsafia — APPROVED** [review]

File: src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs (hunk 1 above)
Comment (eerhardt) [reply src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs:38]: The Grpc template tests are getting updated in that PR already. We can just wait for that PR to get clean.

**Pull request merged**